src/pyclaw/classic/solver.py

```diff
@@ -126,15 +117,30 @@ def step(self,solution):
         :Output: 
          - (bool) - True if full step succeeded, False otherwise
         """
-        
-        self.cfl.set_global_max(0.)
 
-        if self.before_step is not None:
-            self.before_step(self,solution.states[0])
+        # Choose new time step
```

Comment:
It seems like a bad sign that you have taken one block of code and made copies of it now in two places.